### PR TITLE
Clear the cache view on user logout

### PR DESCRIPTION
**Pull request by grafixeyehero**
**Changes**
* Reset cached views on user logout  

**Issues**
https://github.com/jellyfin/jellyfin-web/issues/5615

### CODE REVIEW

File: src/components/ServerConnections.js [hunk 1]
```diff
@@ -40,10 +39,10 @@ class ServerConnections extends ConnectionManager {
 
         Events.on(this, 'localusersignedout', (_e, logoutInfo) => {
             setUserInfo(null, null);
+            // Reset cached views
+            document.dispatchEvent(new CustomEvent('skinunload', {}));
```
Comment (thornbill) [inline src/components/ServerConnections.js:43]: I would remove the code related to that event and just call the `viewContainer` function directly.

```suggestion
            viewContainer.reset();
```

File: src/components/ServerConnections.js [hunk 2]
```diff
@@ -40,10 +39,10 @@ class ServerConnections extends ConnectionManager {
...
             // Ensure the updated credentials are persisted to storage
             credentialProvider.credentials(credentialProvider.credentials());
-            // Reset the query cache
-            queryClient.resetQueries();
```
Comment (thornbill) [inline src/components/ServerConnections.js:46]: Let's move this back since it wasn't the root of the issue and we can keep this reset logic in one place.

**Review by thornbill — CHANGES_REQUESTED** [review]

**Review by scampower3 — APPROVED** [review]

File: src/components/ServerConnections.js (hunk 2 above)
Comment (grafixeyehero) [reply src/components/ServerConnections.js:46]: the `queryClient.resetQueries()` doesn't rest the `tanstack cache`. moving the reset logic to dashboard page > logout it work as expected

Comment (grafixeyehero) [reply src/components/ServerConnections.js:46]: Before

https://github.com/jellyfin/jellyfin-web/assets/32230989/b27b902a-5cc9-4b29-aeea-fe78b58000f2

After

https://github.com/jellyfin/jellyfin-web/assets/32230989/c6723364-f4aa-471b-958a-6a7e294fc94a

**Review by thornbill — APPROVED** [review]

File: src/components/ServerConnections.js (hunk 2 above)
Comment (grafixeyehero) [reply src/components/ServerConnections.js:46]: https://github.com/jellyfin/jellyfin-web/pull/5636

**Pull request merged**